### PR TITLE
Add additional docs in retrieval agent if required

### PR DESCRIPTION
**Pull request by ShobhitVishnoi30**
## Why are these changes needed?

If you have existing documents in the retriever agent instance, and you want to add more documents using the current implementation, it currently replaces the existing ones because the document IDs always start with "doc_0."

To address this, we've updated the functionality to allow assigning different IDs if the user wishes to add additional documents.I've incorporated an "extra_docs" configuration within the retrieval settings. When set to true, the system will assign IDs to the chunks starting from "length+i" when inserting the document into the Chroma database. This modification ensures that existing documents are not replaced, allowing for the addition of more documents to the same collection.

## Related issue number

[issue](https://github.com/microsoft/autogen/issues/1027)

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by ShobhitVishnoi30** [comment]
@sonichi please review it

File: autogen/agentchat/contrib/retrieve_user_proxy_agent.py [hunk 1]
```diff
@@ -171,6 +171,7 @@ def retrieve_docs(self, problem: str, n_results: int = 20, search_string: str =
         self._task = self._retrieve_config.get("task", "default")
         self._client = self._retrieve_config.get("client", chromadb.Client())
         self._docs_path = self._retrieve_config.get("docs_path", None)
+        self._extra_docs = self._retrieve_config.get("extra_docs", False)
```
Comment (thinkall) [inline autogen/agentchat/contrib/retrieve_user_proxy_agent.py:174]: Could you also add docstring for it?

File: autogen/retrieve_utils.py [hunk 2]
```diff
@@ -250,6 +250,7 @@ def create_vector_db_from_dir(
     custom_text_split_function: Callable = None,
     custom_text_types: List[str] = TEXT_FORMATS,
     recursive: bool = True,
+    extra_docs: bool = False,
```
Comment (thinkall) [inline autogen/retrieve_utils.py:253]: Also docstring here?

**Review by thinkall — COMMENTED** [review]
Thanks @ShobhitVishnoi30 for the PR!

Would you like to update the qdrant agent as well?

File: autogen/agentchat/contrib/retrieve_user_proxy_agent.py (hunk 1 above)
Comment (svrapidinnovation) [reply autogen/agentchat/contrib/retrieve_user_proxy_agent.py:174]: sure I will do that

File: autogen/retrieve_utils.py (hunk 2 above)
Comment (svrapidinnovation) [reply autogen/retrieve_utils.py:253]: this is done

File: autogen/agentchat/contrib/retrieve_user_proxy_agent.py (hunk 1 above)
Comment (svrapidinnovation) [reply autogen/agentchat/contrib/retrieve_user_proxy_agent.py:177]: this is done

**Comment by svrapidinnovation** [comment]
> Thanks @ShobhitVishnoi30 for the PR!
> 
> Would you like to update the qdrant agent as well?

Hi @thinkall , added the same in qdrant too

**Review by qingyun-wu — APPROVED** [review]

**Pull request merged**